pyart/retrieve/tests/compare_kdp_proc.py

```diff
@@ -0,0 +1,192 @@
+         1.60713043e+01,   1.26680298e+01,   1.43690491e+01,
+         1.67355728e+01,   1.33732071e+01,   1.49895935e+01,
+         1.45354385e+01,   1.18312225e+01,   1.37792435e+01,
+         1.63132858e+01,   1.49929428e+01,   1.35270767e+01,
+         1.62972488e+01,   1.49179840e+01,   1.16152649e+01,
+         1.37163849e+01,   1.39367752e+01,   1.43221207e+01,
+         1.31950226e+01,   1.61372986e+01,   1.45505676e+01,
+         1.70516205e+01,   1.59943848e+01,   1.62334900e+01,
+         1.43479309e+01,   1.48412476e+01,   1.56809921e+01,
+         1.69597702e+01,   1.23267288e+01,   1.73257904e+01,
+         1.74552383e+01,   1.61041946e+01,   1.59116135e+01,
+         1.67083588e+01,   1.73401337e+01,   1.30528488e+01,
+         1.82811661e+01,   1.89352417e+01,   1.67904205e+01]
+         ])
+         
+    psidp = np.ma.array(psidp)
+        
+    radar = sample_objects.make_empty_ppi_radar(len(psidp[0]), 1, 1)  
+    
+    psidp_dict = {
+        'data': psidp,
+        }
+    radar.add_field(get_field_name('differential_phase'), psidp_dict) 
+    
+    # Define real ranges
+    radar.range['data'] = 75*np.arange(0,len(psidp[0]))
+    return radar
+
+
+def compare_kdp_estimation_methods():
+    # Get profile of noisy psidp
+    prof_psidp = _make_real_psidp_radar()
+    
+    # Maesaka method
+    kdp_dict_mae, phidpf_dict_mae, phidpr_dict_mae = kdp_proc.kdp_maesaka(
+       prof_psidp, maxiter=1000, check_outliers=False)
+    
+    # Vulpiani method (note windsize is just a guess here..)
+    kdp_dict_vulp, phidp_dict_vulp = kdp_proc.kdp_vulpiani(prof_psidp,
+                                              windsize=30, n_iter = 20, band = 'X')
+    
+    # Kalman filter method
+    kdp_dict_schnee,kdp_std_dict_schnee,phidp_dict_schnee = \
+                        kdp_proc.kdp_schneebeli(prof_psidp, band = 'X')
+                
+ 
+    plt.figure(figsize=(10,10))
+    plt.subplot(2,1,1)
+    plt.grid(True)
+    plt.title('Kdp estimation')
+    plt.hold(True)
+    
+    ranges = prof_psidp.range['data']
+    
+    plt.plot(ranges,kdp_dict_mae['data'][0])
+    plt.plot(ranges,kdp_dict_vulp['data'][0])
+    plt.plot(ranges,kdp_dict_schnee['data'][0])
+    plt.xlabel('Range [m]')
+    plt.ylabel(r'Kdp [deg/km]')
+    plt.legend(['Maesaka','Vulpiani','Schneebeli'],loc=0)
+    
+    plt.subplot(2,1,2)
+    plt.grid(True)
+    plt.title('Reconstructed Phidp')
+    plt.hold(True)
+    
+    ranges = prof_psidp.range['data']
+    
+    phidp_dict_mae = 0.5 * (phidpf_dict_mae['data'][0] + 
+                            phidpr_dict_mae['data'][0])
+    plt.plot(ranges,phidp_dict_mae)
+    plt.plot(ranges,phidp_dict_vulp['data'][0])
+    plt.plot(ranges,phidp_dict_schnee['data'][0])
+    plt.plot(ranges,prof_psidp.fields['differential_phase']['data'][0])
+    
+    plt.xlabel('Range [m]')
+    plt.ylabel(r'Diff. phase [deg]')
+    plt.legend(['Maesaka','Vulpiani','Schneebeli','Real Psidp'],loc=0)    
```

Comment:
Would plt.show be helpful here to bring up a window with the plot when this is run from the console?